Build/Build.cs

```diff
@@ -32,14 +33,14 @@ class Build : NukeBuild
 
     public static int Main() => Execute<Build>(x => x.SpellCheck, x => x.Push);
 
-    [Parameter("A branch specification such as develop or refs/pull/1775/merge")]
-    readonly string BranchSpec;
+    GitHubActions GitHubActions => GitHubActions.Instance;
```

Comment:
🤔 @jnyrup do we want to tie ourselves to Github Actions like this? It kind of defies the idea of an independent build script. But we do assume there's a Github branch.

Reply:
My intention was, that we get all variables free of charge just by using this injection.

Locally this won't make a difference by now, because we have default values in this cases (or we do not need them locally)

Reply:
I like the less yaml fiddling and extra type safety.
It seems to have caught that https://github.com/fluentassertions/fluentassertions/commit/c56b55b7f08e9655fece50d3981495a2657956d5 accidentally added `BaseRef` in `.github/workflows/build.yml` for the linux/mac jobs.

While it does add coupling to a particular build system, namely GithubActions, I'm not sure it's _that bad_ because:
* It still runs locally
* The usage of GithubActions is limited to a single file.
* Should we ever want to change to AppVeyor, Azure DevOps, etc, it might be even easier to transition due to intellisense (given NUKE supports it), than adapting a yaml file to a new build system.

... _pendantically_ speaking we also have a coupling to git even though other source code control systems exists ;)

Reply:
> It seems to have caught that https://github.com/fluentassertions/fluentassertions/commit/c56b55b7f08e9655fece50d3981495a2657956d5 accidentally added BaseRef in .github/workflows/build.yml for the linux/mac jobs.

IMO that fail to remove this variable shows the pitfall perfectly..

BTW: this is removed via this PR

Reply:
> Should we ever want to change to AppVeyor, Azure DevOps, etc, it might be even easier to transition due to intellisense (given NUKE supports it), than adapting a yaml file to a new build system.

If we would use the yaml generation part of NUKE, it would be even easier, because we only have to look at the attribute and would get a yaml which is correct for the "new" system

Reply:
OK. Let's keep the specific Github Actions stuff